### DISCUSSION TITLE
Need help identifying lens with IR pass filter for PS3 eye

### DISCUSSION THREAD

**Discussion by tchoopojcharoen** [jyjblrd/Mocap-Drones#37] [Q&A]
Could someone please assist me in identifying a compatible lens for the PS3 camera, preferably with a mount that fits the PS3 Eye PCB? I found information on the blog suggesting that the lens can be replaced entirely.https://joshuabird.com/blog/post/mocap-drones.

**Comment by jyjblrd** [comment]
You would probably need to model the pcb mount yourself and 3d print it. Should be relatively simple, but if you are not comfortable doing that I would suggest just going with the other option of modifying the original camera lens.

You can find a lot of lenses that would probably work if you look up "M12 camera lens". Just make sure to choose a lens with an adequately large FoV (and don't forget to take into account the size of the PS3 camera sensor). Once again, if it seems a bit too complicated I would recommend simply modifying the original lens.

**Comment by gumby0q** [comment]
Hi,
@tchoopojcharoen 
I have had such experience of replacing original lenses recently.

What I did:
- 3d print model for lens mounting can be found here https://github.com/reiserlab/Component-Designs/blob/main/Miscellaneous/Lens-mount_M12/production/PS-eye_lens-mount_M12.stl
it has mirrored pin guides, cut them and it will fit or just buy holder from aliexpress (m12 with mounting screws distance of 18mm)
- if you make 3d printed mounting you will need a tool for making threads: Tap M12x0.5. I just bought it from local store https://krivoyrog.prom.ua/ua/p227556703-mitchik-mashinno-ruchnij.html
- lenses: the more or less appropriate lens is M12 with focal distance of 2.8mm or less (you will get more wide angle, but bigger distortion)
I also bought them from local store https://tehbezpeka.ua/p108997168-obektiv-dlya-kamer.html
- then mount lens, adjust focus and stick lens with hot glue or screw to be fixed in the mounting

some photo (lens on the photo with focal distance of 8.0mm which is to far!)
![photo_2024-05-07_12-28-40](https://github.com/jyjblrd/Low-Cost-Mocap/assets/19606808/54bd7597-f2e8-4864-b951-a538be4d84fc)
![photo_2024-05-07_12-28-54](https://github.com/jyjblrd/Low-Cost-Mocap/assets/19606808/e886f0f7-0f8c-4c45-bb6a-1e8fe91298de)
![photo_2024-05-07_12-31-32](https://github.com/jyjblrd/Low-Cost-Mocap/assets/19606808/626d8768-9da1-4708-8b91-722de76dad34)
 

got I really nicer results and more sharp image with not original lenses

UPDATE: From the datasheet we can assume that original pseye lens has focal length is 2.1 mm. I can confirm that more or less, I was playing with bunch of other lenses near that value. But from my perspective lens with focal distance of 1.8 mm has too much distortion
https://www.playstation.com/content/dam/global_pdc/en/corporate/support/manuals/accessories/ps3-accessories/sceh-00448-ps-eye/SCEH-00448_PS%20Eye_IM%24en.pdf 

But I do not know which values should be written to camera params json yet :(

**Reply by jyjblrd** [reply]
Thanks, this will be very useful for others!

You can follow these instructions to get the camera intrinsic matrix and distortion parameters: [https://docs.opencv.org/4.x/dc/dbb/tutorial_py_calibration.html](https://docs.opencv.org/4.x/dc/dbb/tutorial_py_calibration.html)

**Reply by gumby0q** [reply]
@jyjblrd thank you!

I also left a few points about camera calibration in that thread
https://github.com/jyjblrd/Low-Cost-Mocap/discussions/11

**Reply by luislglp** [reply]
Hey @gumby0q, many thanks for your hints.

So I printed the support, and I did not even needed to make the thread.

Indeed the 2.8 mm is a good option, I was able to adjust the best in focus position by lowering it through the thread while running a code to check the image. 

Here are some photos. But unfortunately I think I bought the wrong one... it says 3MP 1/2.5" IR... From what I understood it actually has the IR filter that we are supposed to remove =(
https://pt.aliexpress.com/item/1005005364717860.html?spm=a2g0o.order_list.order_list_main.17.2b08caa4FQo7Ua&gatewayAdapt=glo2bra

But this focal length and support are really good

![image](https://github.com/jyjblrd/Low-Cost-Mocap/assets/127667739/f7b40adf-b433-49fd-9004-5f04fcb66616)

Update:
Actually it is working, I just added the flopy disk filter to the camera.

<img width="226" alt="image" src="https://github.com/jyjblrd/Low-Cost-Mocap/assets/127667739/3a0fa97a-6a9f-4e4b-854a-b77b51083466">

So I have printed this file, but not sure if I inverted the image but the pins that connected to the PCB were not in the right position, so I removed them. Anyway the screws were enough to hold the base to the PCB.

https://github.com/reiserlab/Component-Designs/blob/main/Miscellaneous/Lens-mount_M12/production/PS-eye_lens-mount_M12.stl

**Reply by luislglp** [reply]
More updates:
Here is the final version....the design is not so clean to be honest.

<img width="770" alt="image" src="https://github.com/jyjblrd/Low-Cost-Mocap/assets/127667739/2e1c8145-0f13-4f3a-b949-ea7395228e6b">

A few details:

<img width="1195" alt="image" src="https://github.com/jyjblrd/Low-Cost-Mocap/assets/127667739/fc75be20-adb9-483c-a7c5-7eaae65e7818">

If anyone requires, detailed product specs can be found here: http://www.feyondcctvled.com/dot-lamp-board/fy-0524.html

So I calibrated each camera without the floppy disk filter.... So I assume that the camera intrinsics would not change after  installing it. To be honest not sure if this is a good assumption since as we can see in the photo it is difficult to install the filter without any distortion (wave like pattern). 

Some lessons learned / suggestions:
- It would be interesting to have a 3D printed attachment on top of the lens that it would be necessary only to slide the filter inside it. 
- Preferable a square size to avoid further distortions if the floppy disk filter is cutted in a round pattern
- Maybe is better to chose on IR floodlight ring that does not have the CDS option.

Currently I am waiting for the passive markers to arrive, the system is working fine with IR LED target. But I still need to check the camera pose calibration that it is not working properly for me... future steps....

**Comment by protocol113** [comment]
The link for the printed lens housing is 404 now, does someone have a link that is still working? @gumby0q if you've still got the file.

**Reply by gumby0q** [reply]
I dug some internet, guy moved the repo or something like that.
Anyway it's kind of bad one
https://github.com/reiserlab/Fly-Lab-Gear/blob/main/Miscellaneous/Lens-mount_M12/production/PS-eye_lens-mount_M12.stl

just buy on ali
https://www.aliexpress.com/item/1005006567692510.html?spm=a2g0o.order_list.order_list_main.5.21ef1802BWFjhw
height 12mm

there are bigger screw holes. - use screws from ps eye case, but for that make pcb holes a bit bigger
<img width="1470" height="621" alt="image" src="https://github.com/user-attachments/assets/7e01b973-af08-4774-a167-0049ad90e0f0" />
